TestProviders/TestProvider.psm1

```diff
@@ -32,7 +32,7 @@ Class TestProvider
 	[string] $CustomLIS
 	[bool]   $ReuseVmOnFailure = $false
 
-	[object] DeployVMs([xml] $GlobalConfig, [object] $SetupTypeData, [object] $TestCaseData, [string] $TestLocation, [string] $RGIdentifier, [bool] $UseExistingRG, [string] $ResourceCleanup) {
+	[object] DeployVMs([xml] $GlobalConfig, [object] $SetupTypeData, [object] $TestCaseData, [string] $TestLocation, [string] $RGIdentifier, [bool] $UseExistingRG, [string] $ResourceCleanup, [switch] $EnableTelemetry) {
```

Comment:
Will the interface change affect other internal providers?
Can you make the EnableTelemetry a property of TestProvider instead?